backend/infrahub/core/query/node.py

```diff
@@ -157,7 +157,10 @@ async def query_init(self, db: InfrahubDatabase, **kwargs) -> None:  # noqa: ARG
         relationships: list[RelationshipCreateData] = []
         for rel_name in self.node._relationships:
             rel_manager: RelationshipManager = getattr(self.node, rel_name)
+            peers = await rel_manager.get_peers(db=db, branch_agnostic=self.branch_agnostic)
             for rel in rel_manager._relationships:
+                if rel.get_peer_id() in peers:
+                    await rel.set_peer(value=peers[rel.get_peer_id()])
```

Comment:
```suggestion
for rel in rel_manager.get_relationships(db=db, branch_agnostic=self.branch_agnostic):
```

This should fetch all relationships at once, and it avoids accessing private members `_relationships`

Reply:
I already tried that and the thing is that `get_relationships()` only sets the peer ids (and does not call a get_peers equivalent to fetch the full peer node).

I wonder if we really need the peer attributes for the NodeCreateAll query to work, maybe we can just avoid the `peer.resolve()` call within the `rel.get_create_data` function.

Reply:
>I already tried that and the thing is that get_relationships() only sets the peer ids (and does not call a get_peers equivalent to fetch the full peer node).

Right 

It seems we need to fetch to get the branch peer